added code for optimised training of KNN on single images and all images

sreenithi opened the pull request.

sreenithi committed:
added code for optimised training of KNN on single images and all images

gaganmanku96 commented on .vscode/settings.json:
```diff
@@ -0,0 +1,3 @@
+{
+    "python.pythonPath": "C:\\ProgramData\\Anaconda3\\python.exe"
```
please add .vscode folder in .gitignore

gaganmanku96 committed:
added new images to verify the model predictions

gaganmanku96 commented:
@sreenithi I have pushed some new images in your branch. new_labels is giving me 2 unique classes and if I'm not wrong there should be 4 unique classes (after new images have been added). Can you confirm once?

Thank you

sreenithi committed:
added prints and fixed grouped image names

sreenithi commented:
Hi @gaganmanku96 
There were 4 unique classes actually, but the print statements were displayed only for those classes for which similar images were found, which are class 0 (Shahrukh pictures) and class 1 (Aishwarya pictures).
I've added some extra prints to show all the 4 classes and images in each classes now, and also fixed a small bug in the grouped image names which I had not noticed yesterday.

gaganmanku96 commented:
@sreenithi I couldn't understand why you have set n_neighbors in full_knn as classNum-1. Isn't it supposed to be classNum i.e 4?

sreenithi commented:
@gaganmanku96 Sorry, my bad. It is supposed to be classNum only. Initially, I had considered classNum to start from 1 instead of 0, that's why I had it as classNum-1. Later I had changed it to 0 but forgot to change it here

sreenithi committed:
fixed n_neighbors in full data training
